Swift 2.0 compatibility

gfontenot opened the pull request:
This is going to be merged into `master` soon, but I wanted to open a PR
to track the changes. Most of these have been given a :+1: in their
individual PRs (#37, #38, #40), but they will be collected here as well.

Commit:
Add watchOS target

There is no test target for watch frameworks yet so there is not
associated test target.

sync committed:
Use named generic types for Arrays

In Xcode 7 beta 3, they renamed the generic types to be more
descriptive. This change makes us compatible with that change.

JaviSoto committed:
Share watchOS scheme

gfontenot committed:
Update dependencies

gfontenot committed:
Update project file for Xcode 7

gfontenot committed:
Update documentation for Swift 2.0/Xcode 7

gfontenot committed:
Update `compose` for new naming requirements

Argument labels have been made more consistent in 2.0, which is awesome,
but we don't need the second label here.

gfontenot committed:
Make `apply` instance methods public

Swift 2.0 lets us create public extensions on generic types again!
Huzzah! That means we can publicize `apply`, which will be nice for when
using an instance method makes more sense than using an operator.

This is something I always _wanted_ to do, but was never _able_ to do.
Pumped to make this public.

gfontenot committed:
Point to 2.0 release of NimbleFox

gfontenot committed:
Use versioning for Quick requirement

We should make sure that we're pointing to the proper version of Quick
here. It wasn't great that we weren't doing this already.

gfontenot commented:
@thoughtbot/ios I think this is good now. The main change here is that we're now publicizing the `apply` instance method for `Array` and `Optional`. The rest is just source/Xcode compatibility and adding a watchOS framework.

gfontenot committed:
Update README to indication Swift 2.0 compatibility

tonyd256 commented on Runes.xcodeproj/project.pbxproj:
```diff
@@ -508,6 +613,7 @@
 				COPY_PHASE_STRIP = NO;
 				CURRENT_PROJECT_VERSION = 1;
 				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				ENABLE_TESTABILITY = YES;
```
yay

sidraval commented on Source/Array.swift:
```diff
@@ -57,25 +57,25 @@ public func -<<<T, U>(@noescape f: T -> [U], a: [T]) -> [U] {
+        - returns: A value of type [U]
     */
-    func apply<U>(fs: [T -> U]) -> [U] {
+    func apply<U>(fs: [Element -> U]) -> [U] {
```
If we're going to call this `Element` here, should we call it that in the `- parameter fs:` above?

gfontenot replied:
Yeah, good catch. Maybe we should change _all_ of these to `Element`? Does `U` make much sense now? I'm not sure what else to call that...

sidraval replied:
I actually prefer `T` to `Element`, did it change because there's no way to indicate the type of the elements in an Array extension?

gfontenot replied:
I agree, I'll leave `T` for the operators, but will change the documentation to match this.

Swift changed these types to be more semantic than just using `T`. It makes sense in _most_ contexts, but gets weird here.

gfontenot committed:
Change type name from T -> Element in docs

This is for the sake of consistency

gfontenot committed:
Update documentation with code blocks

Since Swift 2.0 uses markdown for documentation parsing, we can use
backticks to indicate code blocks. Which is neat.

sidraval commented:
:+1:

gfontenot commented:
This is merged into master now. I'm going to leave this branch up for a bit because of external people pointing to it, but you should use the `3.0.0-rc.1` tag moving forward.
